internal/campaigns/logger.go

```diff
@@ -242,6 +242,38 @@ func (a *ActionLogger) DockerStepDone(repoName string, step int, elapsed time.Du
 	a.write(repoName, yellow, "%s Done. (%s)\n", boldBlack.Sprintf("[Step %d]", step), elapsed)
 }
 
+func (a *ActionLogger) RepoMatches(repoCount int, unsupported []string) {
+	unsupportedCount := len(unsupported)
+	var matchesStr string
+	if repoCount == 1 {
+		matchesStr = fmt.Sprintf("%d repository matches the scopeQuery.", repoCount)
+	} else {
+		var warnStr string
+		if repoCount == 0 {
+			warnStr = "WARNING: "
+		}
+		matchesStr = fmt.Sprintf("%s%d repositories match the scopeQuery.", warnStr, repoCount)
+	}
+	if unsupportedCount > 0 {
+		matchesStr += fmt.Sprintf(" (Including %d on unsupported code hosts.)", unsupportedCount)
+	}
+	if unsupportedCount > 0 {
+		matchesStr += "\n\nThe following repositories were filtered out because they are on a codehost not supported by campaigns (Use -include-unsupported to generate patches for them anyways.):\n"
```

Comment:
Slight grammar nit: `anyways` should be singular in formal contexts. I also find the full sentence a little distracting in the parentheses, so I'd be inclined to make it a sentence fragment like the below:

```suggestion
		matchesStr += "\n\nThe following repositories were filtered out because they are on a codehost not supported by campaigns (use -include-unsupported to generate patches for them anyway):\n"
```